logs/util/read_logs_blocker.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package util
+
+import (
+	"sync"
+)
+
+var logBlockerSingleton *LogBlocker
+
+type LogBlocker struct {
+	maxLogBufferSize int64
+	logsBufferSize int64
+	logsBufferSizeMutex sync.Mutex
+}
+
+func GetLogBlocker() *LogBlocker {
+	if logBlockerSingleton == nil {
+		logBlockerSingleton = &LogBlocker{
+			maxLogBufferSize: int64(-1),
+			logsBufferSize: 0,
+			logsBufferSizeMutex: sync.Mutex{},
+		}
+	}
+	return logBlockerSingleton
+}
+
+func (l *LogBlocker) Add(v int64) {
+	l.logsBufferSizeMutex.Lock()
+	defer l.logsBufferSizeMutex.Unlock()
+	l.logsBufferSize = l.logsBufferSize + v
+}
+
+func (l *LogBlocker) Subtract(v int64) {
+	l.logsBufferSizeMutex.Lock()
+	defer l.logsBufferSizeMutex.Unlock()
+	l.logsBufferSize = l.logsBufferSize - v
+}
+
+// Reset this is test only code
+// do not use for application
+func (l *LogBlocker) Reset() {
```

Comment:
I'm not sure how to get around having this here, even though this should be for testing only

Reply:
I'm not certain how to deal with test side effects. Since this is a singleton instance.

Reply:
The problem is relying on a global singleton instead of using composition so we could replace it with a test log blocker.

Reply:
Converted to a struct that is passed down to all required places.